README.md

```diff
@@ -1 +1,75 @@
-# snowflake-connector
\ No newline at end of file
+# Snowflake REST API connector #
+
+Snowflake is a cloud data warehouse that can store and analyze all your data records in one place. It can automatically scale up/down its compute resources to load, integrate, and analyze data.
+
+The Kosha connector enables you to perform REST API operations on the Snowflake API in your Kosha workflow or custom application. Using the Kosha-Snowflake connector, you can:
+
+* Submit SQL statements for execution.
+* Check the status of the execution of a statement.
+* Cancel the execution of a statement.
+
+## Authentication ##
+
+Kosha uses OAuth 2.0 to connect to Snowflake. If you already have an account with Snowflake, you can use your 'userid' and 'password' associated with your account along with your 'account_identifier' when provisioning the connector.
+
+If you don’t want to use those credentials when provisioning the Snowflake connector, Kosha provides bootstrap credentials. After you sign in to your Snowflake account, Snowflake gives Kosha an access token and your connector is provisioned. Kosha automatically refreshes your access token before it expires to ensure there’s no disruption in use.
+
+
+Once you've logged-in to your account, you would need to authenticate to the server when using the Snowflake SQL API.
+
+### Enabling OAuth Connection ###
+
+1. The user would first have to set the default account role to sysadmin. This can be achieved by the below query. 
+```alter user admin set default role = sysadmin;```
+
+2. The user would also have to create a database using the query ```CREATE DATABASE <enter database-name>;``` and then
+choose an existing warehouse from the list of warehouses in their account 
+```show warehouses;``` or create a new one ```CREATE OR REPLACE WAREHOUSE <warehouse-name> WITH WAREHOUSE_SIZE='<enter desired warehouse size>';``` 
+
+3. Now, the user would have to enable OAuth2 for accessing the API. This can be achieved by running the below query.
+```
+create or replace security integration <enter security integration name>
+type = oauth
+enabled = true
+oauth_client = custom
+oauth_client_type = 'CONFIDENTIAL'
+oauth_redirect_uri = 'https://oauth.pstmn.io/v1/callback'
```

Comment:
let's remove the reference to postman and ask the user to enter their own callback (redirect) URI